stellargraph/mapper/sliding.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = [
+    "SlidingFeaturesNodeGenerator",
+    "SlidingFeaturesNodeSequence",
+]
+
+import numpy as np
+from . import Generator
+from tensorflow.keras.utils import Sequence
+
+from ..core.validation import require_integer_in_range
+
+
+class SlidingFeaturesNodeGenerator(Generator):
+    def __init__(self, G, window_size, batch_size=1):
+        require_integer_in_range(window_size, "window_size", min_val=1)
+        require_integer_in_range(batch_size, "batch_size", min_val=1)
+
+        self.graph = G
+
+        node_type = G.unique_node_type(
+            "G: expected a graph with a single node type, found a graph with node types: %(found)s"
+        )
+        self._features = G.node_features(node_type=node_type)
+
+        self.window_size = window_size
+        self._batch_size = batch_size
+
+    def num_batch_dims(self):
+        return 1
+
+    def flow(self, sequence_iloc_slice, target_distance=None):
```

Comment:
It would be good to paste in some of the explanation in the issue description as part of the docstring here for what the arguments are referring to.

In particular, I think it may not be obvious that the query length is calculated after adding the target distance, instead of before? I'm guessing in most cases you'd use this with `slice(None)` to use the whole sequence, but if someone wanted to use a particular slice, e.g. `slice(0, 10)` they might expect the training set to have 10 examples, instead of `10 - target_distance`. Thoughts?

Reply:
Oh, yeah, this whole class lacks documentation. I'll add.